eclass/blas.eclass

```diff
@@ -0,0 +1,241 @@
+# @ECLASS: blas.eclass
+# @MAINTAINER:
+# schmidom@student.ethz.ch
+# @BLURB: Selects a proper BLAS implementation to build the package
+# @DESCRIPTION:
+# Here a BLAS-implementation is chosen from the cutset of available,
+# compatible and implementations in the USE-flags
+# The implementation is then enforced upon every dependency of the package
+
+# @ECLASS-VARIABLE: BLAS_COMPAT
+# @REQUIRED
+# @DESCRIPTION:
+# This variable contains a list of BLAS implementations the package
+# supports. It has to be an array. Either it or BLAS_COMPAT_ALL must be
+# set before the `inherit' call. BLAS_COMPAT_ALL overrides BLAS_COMPAT.
+#
+# Example:
+# @CODE
+# BLAS_COMPAT=( refblas openblas gotoblas mkl )
+# @CODE
+
+# @ECLASS-VARIABLE: BLAS_COMPAT_ALL
```

Comment:
1. Is there a reason to have two variables? The 'standard' practice would be to allow all if `BLAS_COMPAT` is not set. However…
2. Allowing 'all' is problematic by design (this is why python-r1 doesn't do it). It means that whenever a new implementation is supported by the eclass, all packages instantly start depending on it. This means that before adding a new implementation, you need to keyword it on all relevant architectures and test all reverse dependencies.
3. How frequent are BLAS/LAPACK incompatibilities?

Reply:
Incompatibilities are few and far between. You could possibly have trouble with intel MKL because it does more than just provide blas/lapack. It provides FFTW and gsl like functionalities. But so long has you are interested in the published blas/lapack functions defined in the reference package there shouldn't be incompatibilities.

This is why I suggested we follow the multilib model rather than the python model.

Reply:
@mgorny 
1. has been fixed in 
2. "All" means that the program only requires the common functions of BLAS, i.e. everything in refblas. A package that doesn't provide these core-functions is no BLAS implementation.
3. Some implementations have extra functions like openblas_set_num_threads (or similar). If a program uses those, it is incompatible with the other implementations

@kiwifb If a program uses these functions, the eclass probably has to depend on MKL directly.